mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go

```diff
@@ -298,9 +299,13 @@ func resourceBigQueryTableSchemaIsChangeable(old, new interface{}, topLevel bool
 				resourceBigQueryTableSchemaIsChangeable(mapOld[key], mapNew[key], false); err != nil || !isChangable {
 				return false, err
 			}
```

Comment:
I thought we'd be updating this as we look through `mapOld` and comparing key to `mapNew` - could you explain how this is working as intended?

Reply:
not sure I understand the question correctly, we are going through the keys in `mapOld` and comparing to `mapNew`, but we're doing it recursively because of the nature of the schema (nested interfaces). We immediately return false (not changeable) if we encounter an unchangeable case, for example a column that had it's type changed.

Reply:
I was referring to this block:
```
if topLevel {
  sameNameColumns += 1
}
```

At a glance, in a single given level, top or nested, `sameNameColumns` will be 0 or 1. Additional comments on the variables and the logic here would help.

Reply:
`sameNameColumns ` is always 0 for any nested level, and could be 0 or more at the top level. I refactored this part of the code, it should still be the same logic but I think It's more readable now and less prone to accidental logic-breaking changes.